File: engine/tests/fixtures/aggregator_sdk/src/routed.ts
import { createOpenAI } from "@ai-sdk/openai";
import { anthropic } from "@ai-sdk/anthropic";
import { generateText } from "ai";

const openai = createOpenAI({ apiKey: process.env.OPENAI_API_KEY });

export async function viaVercelOpenAI(prompt: string) {
  return generateText({ model: openai("gpt-4-0613"), prompt });
}

export async function viaVercelAnthropic(prompt: string) {
  return generateText({ model: anthropic("claude-3-opus-20240229"), prompt });
}

File: engine/tests/fixtures/aggregator_sdk/src/routed.py
import litellm
from langchain_openai import ChatOpenAI
from langchain_anthropic import ChatAnthropic


def via_litellm_prefixed(messages):
    # Provider is carried by the model string, not the callee.
    return litellm.completion(model="anthropic/claude-3-opus-20240229", messages=messages)


def via_litellm_bare(messages):
    # litellm resolves a bare name through its own routing table.
    return litellm.completion(model="gpt-4-0613", messages=messages)


def via_litellm_dynamic(model, messages):
    # Provider unknown until runtime.
    return litellm.completion(model=model, messages=messages)


openai_chat = ChatOpenAI(model="gpt-4-0613")
anthropic_chat = ChatAnthropic(model="claude-3-opus-20240229")
